scripts/generateAssetData/utils/portals.ts

```diff
@@ -92,20 +129,86 @@ export const getPortalTokens = async (nativeAsset: Asset): Promise<Asset[]> => {
       assetNamespace: chainId === bscChainId ? ASSET_NAMESPACE.bep20 : ASSET_NAMESPACE.erc20,
       assetReference: token.address,
     })
+    const asset = assets[assetId]
+
+    const platform = portalsPlatforms[token.platform]
+    const isPool = Boolean(platform && token.tokens?.length) || undefined
+
+    const iconOrIcons = (() => {
+      // There are no underlying tokens, return asset icon
+      if (!token.tokens?.length) return { icon: asset?.icon }
+
+      // This is a multiple assets pool, populate icons array
+      if (token.tokens.length > 1)
+        return {
+          icons: token.tokens
+            .map(underlyingToken => {
+              const underlyingAssetId = toAssetId({
+                chainId,
+                assetNamespace:
+                  chainId === bscChainId ? ASSET_NAMESPACE.bep20 : ASSET_NAMESPACE.erc20,
+                assetReference: underlyingToken,
+              })
+              const underlyingAsset = assets[underlyingAssetId]
+              return underlyingAsset?.icon
+            })
+            .filter(isSome),
+          icon: undefined,
+        }
+
+      // Only a single asset pool, no need to populate the icons array
+      return { icon: asset?.icon }
+    })()
+
+    // New naming logic
+    const name = (() => {
+      // For single assets, just use the token name
+      if (!isPool) return token.name
+      // For pools, create a name like "ASSET1/ASSET2 Pool"
+      const assetSymbols =
+        token.tokens?.map(underlyingToken => {
+          const assetId = toAssetId({
+            chainId,
+            assetNamespace: chainId === bscChainId ? ASSET_NAMESPACE.bep20 : ASSET_NAMESPACE.erc20,
+            assetReference: underlyingToken,
+          })
+          const underlyingAsset = assets[assetId]
+          if (!underlyingAsset) return undefined
+
+          // This doesn't generalize, but this'll do, this is only a visual hack to display native asset instead of wrapped
+          // We could potentially use related assets for this and use primary implementation, though we'd have to remove BTC from there as WBTC and BTC are very
+          // much different assets on diff networks, i.e can't deposit BTC instead of WBTC automagically like you would with ETH instead of WETH
+          switch (underlyingAsset.symbol) {
+            case 'WETH':
+              return 'ETH'
+            case 'WBNB':
+              return 'BNB'
+            case 'WMATIC':
+              return 'MATIC'
+            case 'WAVAX':
+              return 'AVAX'
+            default:
+              return underlyingAsset.symbol
+          }
+        }) ?? []
+      return `${assetSymbols.join('/')} Pool (${platform.name})`
+    })()
+
+    console.log({ name, address: token.address, iconOrIcons })
```

Comment:
console.log is on purpose while debugging to get a visual overview , remove me before opening this PR